packages/app/src/app/hooks/useScript.ts

```diff
@@ -0,0 +1,60 @@
+// Based on https://usehooks.com/useScript/
+import { useEffect, useState } from 'react';
+
+const cachedScripts = [];
+
+export const useScript = (src: string) => {
+  const [state, setState] = useState({
+    loaded: false,
+    error: false,
+  });
+
+  useEffect(() => {
+    if (cachedScripts.includes(src)) {
+      setState({
+        loaded: true,
+        error: false,
+      });
+    } else {
+      cachedScripts.push(src);
+
+      const script = document.createElement('script');
+      script.src = src;
+      script.async = true;
+
+      const onScriptLoad = () => {
+        setState({
+          loaded: true,
+          error: false,
+        });
+      };
+
+      const onScriptError = () => {
+        const index = cachedScripts.indexOf(src);
+
+        if (index >= 0) cachedScripts.splice(index, 1);
```

Comment:
It will make this piece of code a bit clearer if you wrap it in curly braces (`{}`).

```js
if (index >= 0) {
  cachedScripts.splice(index, 1);
}
```

Reply:
Sounds good!